siakg/main_test.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/NebulousLabs/Sia/modules/tester"
+)
+
+// The tests in this file need to be checked manually, even if errors are
+// returned it will not be apparent to the test suite.
+//
+// CONTRIBUTE: Make these tests automatic.
```

Comment:
you set the error during main, then the testing package can see that it got set to the expected value. This is only necessary b/c main can't return an error.

I'll start with the separate functions, and see how effective that is.